### DISCUSSION TITLE
Programmatic controller with anchor names and navigation history

### DISCUSSION THREAD

**Discussion by fernandogiacomino** [Hejtmus/svelte-fullpage#65] [Q&A] [answered]
Hi! I've found how to navigate with fullpageController and sectionController, however i'm comparing with fullpage.js behaviour and wish to record navigation history and maybe use just one anchor name and not index number. Any help is welcome!

**Comment by Hejtmus** [comment, marked as answer]
Hey, thanks for asking. My answer won't make you happy, but here it is. There is no such a feature as navigation history, so for now if you need this kind of functionality, you would need some complicated solution eg. using intersection observers. 
But if you need some ground for implementing history, I can make quick minor version with option to add callback on `fullpageController` or `sectionController` navigation, which allows you to make wrapper around controllers and thus record navigation states.
And that other thing about anchor name, I'm not familiar with fullpage.js terminology, but if I got you correctly, you want to be able to navigate to section/slide based on it's title, right?

**Reply by fernandogiacomino** [reply]
I like the idea of having some prioritary route, like an array of steps: {section1, section2, slide1, slide2, section3, ...} so you can point a desirable path when rolling mouseweel, even when you can click on points or links to other sections.
I whas reading about window.history api and seems like a local storage where you can send some navigation data, maybe registering there the active section before leave it, always with root url. I'm newbe on this, if don't make any sense just forget it. Thanks!

**Reply by Hejtmus** [reply]
I like the history API approach, it could be done using hash in URL, but that could break pages that use hash based routers (an also sometimes I use this library as slider), so it would be good to be optional feature. That's also reason why I would prefer JS only approach, like you proposed earlier. Feel free to create issue requesting this feature. And thanks for feedback, really appreciate.

**Comment by fernandogiacomino** [comment]
https://github.com/alvarotrigo/fullpage.js Here a great library for fullpage, but yours is more simple and readable.

**Reply by Hejtmus** [reply]
I know it, this library started, when I needed Fullpage.js wrapper for svelte and then realized, that it defeats whole purpose of using svelte on one project, but thanks anyways. Also thanks for noticing simplicity and readability, that was major intention ;).
